src/streams.h

```diff
@@ -398,18 +399,18 @@ class CDataStream
         nReadPos = nReadPosNext;
     }
 
-    void write(const char* pch, size_t nSize)
+    void write(Span<const value_type> src)
     {
         // Write to the end of the buffer
-        vch.insert(vch.end(), pch, pch + nSize);
+        vch.insert(vch.end(), src.data(), src.end());
```

Comment:
For consistency (and compatibility with `std::span`) probably better use `src.begin()` rather than `src.data()` here.